shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -350,10 +358,16 @@ class Scope(val name: Str, enclosing: Opt[Scope]) {
       allowEscape: Bool
   ): StubValueSymbol = {
     val symbol = lexicalValueSymbols.get(lexicalName) match {
+      // If the existing symbol is a value symbol, but the value symbol is
+      // declared in the dry-run of new definition typing, we can reuse the
```

Comment:
What's a dry-run?! Is it explained somewhere?

Reply:
No, I can't think of a name for the process of iterating these symbols for the first time either, but I can explain this in the comments.

Reply:
The whole thing is crazy. I'm not very happy about how @NeilKleistGao just kept piling hacks instead of fixing the problem at the source in a principled way. But this will all be refactored soon anyway.

Reply:
I update the comment and add a link to this PR so unfortunate future reader won't get lost.